sources/HUBViewModelDiff.m

```diff
@@ -57,7 +64,9 @@ + (instancetype)diffFromViewModel:(id<HUBViewModel>)fromViewModel
                         algorithm:(HUBDiffAlgorithm)algorithm
 {
     NSParameterAssert(algorithm);
-    return algorithm(fromViewModel, toViewModel);
+    HUBViewModelDiff *diff =  algorithm(fromViewModel, toViewModel);
+    [diff calculateHeaderChangesFromViewModel:fromViewModel toViewModel:toViewModel];
+    return diff;
```

Comment:
I'm not saying we should change the public convenience constructor, just to introduce the `headerComponentHasChanged` parameter into the internal `initWithInserts:deletes:reloads:` initializer. This way the diffing algorithms would have full responsibility of all the diffing logic, and the `HUBViewModelDiff` would remain a immutable plain old data type.